mkfs/mkfs.c

```diff
@@ -360,22 +361,29 @@ static int exfat_create_root_dir(struct exfat_blk_dev *bd,
 
 static void usage(void)
 {
-	fprintf(stderr, "Usage: mkfs.exfat\n");
-	fprintf(stderr, "\t-L | --volume-label=label                              Set volume label\n");
-	fprintf(stderr, "\t-c | --cluster-size=size(or suffixed by 'K' or 'M')    Specify cluster size\n");
-	fprintf(stderr, "\t-b | --boundary-align=size(or suffixed by 'K' or 'M')  Specify boundary alignment\n");
-	fprintf(stderr, "\t-f | --full-format                                     Full format\n");
-	fprintf(stderr, "\t-V | --version                                         Show version\n");
-	fprintf(stderr, "\t-v | --verbose                                         Print debug\n");
-	fprintf(stderr, "\t-h | --help                                            Show help\n");
+	fputs("Usage: mkfs.exfat\n"
+		"\t-L | --volume-label=label                              Set volume label\n"
+		"\t-c | --cluster-size=size(or suffixed by 'K' or 'M')    Specify cluster size\n"
+		"\t-b | --boundary-align=size(or suffixed by 'K' or 'M')  Specify boundary alignment\n"
+		"\t     --pack-bitmap                                     Move bitmap into FAT segment\n"
+		"\t-f | --full-format                                     Full format\n"
+		"\t-V | --version                                         Show version\n"
+		"\t-v | --verbose                                         Print debug\n"
+		"\t-h | --help                                            Show help\n",
+		stderr);
 
 	exit(EXIT_FAILURE);
 }
 
-static struct option opts[] = {
+enum LongOption {
+	PACK_BITMAP = CHAR_MAX + 1,
+};
```

Comment:
Any reason why you declare enumeration here ? Can not we use just define macro ?

Reply:
I'm a C++ programmer primarily and am habituated to avoiding macros unless I need their definitions at preprocessing time since they aren't type-safe and don't obey scoping rules. I know traditional C programming uses macros for compile-time constants. Feels like an anti-pattern from my C++ perspective, but I'll go with that if that's the prevailing coding style in this project.

Reply:
Yes, exfatprogs have been written with kernel coding style and c programming. In my opinion, It would be better to change it with define macro.

Reply:
Incidentally, the [kernel coding style](https://www.kernel.org/doc/html/latest/process/coding-style.html#macros-enums-and-rtl) guidelines say:

> Enums are preferred when defining several related constants.

So when/if more long-only options are eventually defined, then `PACK_BITMAP` may have to go back to being an enumerator. Worse, because it was once defined as a macro, after it is changed into an enumerator it may need the `#define PACK_BITMAP PACK_BITMAP` trick so that it is still a macro for purposes of `#ifdef`.